### PR TITLE
Initial implementation

### PR DESCRIPTION
**Pull request by pbrisbin**
See `README.md`

### CODE REVIEW

File: src/Rcl/Run.hs [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    <> displayShow (length relevantPackages)
+    <> " are dependencies"
+
+  changes <- for relevantPackages $ \ChangedPackage {..} -> do
+    change <- case cpDiff of
+      This version -> pure $ Removed version
+      That version -> pure $ Added version
+      These fromVersion toVersion ->
+        Changed fromVersion toVersion
+          <$> getChangelogFromTo cpName toVersion fromVersion
```
Comment (pbrisbin) [inline src/Rcl/Run.hs:50]: Need to rename the function or accept the arguments in reverse order.

File: src/Rcl/Options.hs [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+import qualified Data.Set as Set
+import Options.Applicative
+import Rcl.PackageName
+import Rcl.Resolver
+
+data Options = Options
+  { oDebug :: Bool
+  , oFromResolver :: Resolver
+  , oToResolver :: Resolver
+  , oDependencies :: Set PackageName
```
Comment (eborden) [inline src/Rcl/Options.hs:21]: I would be awesome if you could use the `stack.yaml` to source `package.yaml` files and pull dependencies from there. Shouldn't be too hard to do.

File: src/Rcl/ChangedPackage.hs [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+  { comparing :: [Text]
+  , diff :: HashMap PackageName (HashMap Resolver Version)
+  }
+  deriving stock Generic
+  deriving anyclass FromJSON
+
+getResolverDiff :: Resolver -> Resolver -> RIO App [ChangedPackage]
+getResolverDiff fromResolver toResolver = do
+  req <-
+    parseRequestThrow
```
Comment (eborden) [inline src/Rcl/ChangedPackage.hs]: Very convenient that this exposes JSON.

**Review by eborden — APPROVED** [review]

File: src/Rcl/Run.hs [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+    logDebug $ "Changed " <> displayShow cpName <> ": " <> displayShow change
+    pure $ renderChanged cpName change
```
Comment (eborden) [inline src/Rcl/Run.hs:53]: I'd love if the output included a link to the stackage diff page.

File: src/Rcl/Options.hs (hunk 2 above)
Comment (pbrisbin) [reply src/Rcl/Options.hs:21]: I did start with that intention, but as soon as you need to be generalized or flexible to possible project setups, it quickly spirals. It was *much* easier to leave it to the caller to do something like this: https://github.com/freckle/megarepo/blob/master/backend/scripts/dependencies

Comment (pbrisbin) [reply src/Rcl/Options.hs:21]: I could add this small shell snippet to the `README` though.

**Pull request merged**